test/models/policy_host_test.rb

```diff
@@ -5,4 +5,65 @@
 class PolicyHostTest < ActiveSupport::TestCase
   should belong_to(:policy)
   should validate_presence_of(:host)
+
+  setup do
+    account1 = FactoryBot.create(:user).account
+    account2 = FactoryBot.create(:user).account
+    v1 = SupportedSsg.by_os_major['7'].first
+    bm1 = FactoryBot.create(
+      :benchmark,
+      version: v1.version,
+      os_major_version: '7'
+    )
+
+    v2 = SupportedSsg.by_os_major['7'].last
+    bm2 = FactoryBot.create(
+      :benchmark,
+      version: v2.version,
+      os_major_version: '7'
+    )
+
+    @policy1 = FactoryBot.create(:policy, account: account1)
+    @policy2 = FactoryBot.create(:policy, account: account2)
+    parent_profile1 = FactoryBot.create(:canonical_profile, upstream: false, benchmark: bm1)
+    parent_profile2 = FactoryBot.create(:canonical_profile, upstream: false, benchmark: bm2)
+    profile1 = FactoryBot.create(:profile, parent_profile: parent_profile1, policy: @policy1, account: account1)
+    profile2 = FactoryBot.create(:profile, parent_profile: parent_profile2, policy: @policy2, account: account2)
+    profile1.update!(ref_id: parent_profile1.ref_id)
+    profile2.update!(ref_id: profile2.ref_id)
```

Comment:
You don't need to create and set the canonical profile, it all generates automatically. Just pass the benchmarks to the profiles below.